modules/dnn/src/dnn_utils.cpp

```diff
@@ -45,17 +45,18 @@ void getMat(Mat& blob, InputArray blob_, AccessFlag flag) {
     }
 }
 
-void makeMatFromBlob(Mat& m, InputArray blob, int i, int j, int rows, int cols, int type) {
+void getChannelFromBlob(Mat& m, InputArray blob, int i, int j, int rows, int cols, int type) {
     m = Mat(rows, cols, type, blob.getMat().ptr(i, j));
 }
 
-void makeMatFromBlob(UMat& m, InputArray blob, int i, int j, int rows, int cols, int type) {
+void getChannelFromBlob(UMat& m, InputArray blob, int i, int j, int rows, int cols, int type) {
     UMat ublob = blob.getUMat();
-    int offset = i * cols + j;
+    int offset = (i * ublob.step.p[0] + j * ublob.step.p[1]) / ublob.elemSize();
```

Comment:
I forgot to take into account step() and elemSize() for the offset. Now it works